paper/scripts/run_matching.py

```diff
@@ -0,0 +1,406 @@
+        info = pickle.load( open(  path_saving_model+"conf.p", "rb" ) )
+        initial_run = info['run_num']
+        run_acc = info['accuracies']
+    for run in range(initial_run,config['runs']):
+        run_time = time.time()
+        utils.set_seed(run)
+        torch.cuda.init()
+        model_encoder = utils.get_model(config['model'],num_classes,model_type=modality)
+        model_encoder = model_encoder.to(device)
+        num_input_features = model_encoder.linear.in_features
+        model_encoder.linear = torch.nn.Identity()
+        fce = True
+        model = MatchingNetwork(model_encoder, fce, num_input_channels=3, lstm_layers=1,
+                        lstm_input_size=num_input_features, unrolling_steps=2, device=device)
+
+        # training parameters
+        if FLAGS.optimizer == 'sgd':
+            optimizer = torch.optim.SGD(model.parameters(),**dict_optim)
+        elif FLAGS.optimizer == 'adam':
+            optimizer = torch.optim.Adam(model.parameters(),
+                                lr=float(config['optimizer']['learning_rate']))
+        else:
+            raise ValueError("Optimizer not supported")
+        if dataset_name == 'CINIC10':
+             scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, 
+                                T_max=config[dataset_name]['num_epochs'])
+        else:
+            scheduler = torch.optim.lr_scheduler.MultiStepLR(optimizer,  milestones=opt_milestones)
+        
+        # get dataset
+        train_loader, _, test_loader, mem_loader = utils.get_loaders(config,run,balanced=False)
+        train_time = time.time()
+        model = train_model(model, [train_loader,mem_loader], optimizer, scheduler,
+                        loss_criterion, config[dataset_name]['num_epochs'], device=device) 
+        init_eval_time = time.time()
+        best_acc, _ = eval(model,test_loader, mem_loader,loss_criterion,device)
+        end_eval_time = time.time()
+
+        # stats
+        run_acc.append(best_acc)
+
+        # save
+        if save and path_saving_model:
+            saved_name = "{}.pt".format(run+1)
+            save_path = os.path.join(path_saving_model, saved_name)
+            torch.save({'model_state_dict':model.state_dict(),
+            'train_examples': config['train_examples'],
+            'mem_examples':  config[config['dataset_name']]['mem_examples'],
+            'model_name': config['model'],
+            'num_classes': num_classes, 'modality':modality, 'dataset_name':config['dataset_name']} , save_path)
+            info = {'run_num':run+1,'accuracies':run_acc}
+            pickle.dump(info, open( path_saving_model+"conf.p", "wb" ) )
+
+        # log
+        print(f"Run:{run+1} |  Accuracy {best_acc:.2f} | " \
+              f"Mean Accuracy:{np.mean(run_acc):.2f} | Std Dev Accuracy:{np.std(run_acc):.2f}\t" \
+              f"T:{(train_time -run_time)/60:.2f}min\tE:{(end_eval_time -init_eval_time)/60:.2f}")
+
+
+
+
```

Comment:
This is a killer

Reply:
fixed in [5f91fc9](https://github.com/KRLGroup/memory-wrap/pull/3/commits/5f91fc9eb377039c0901309f0346f6cf4d70788b)